drivers/counter/counter_dw_timer.c

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT snps_dw_timers
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/logging/log.h>
+
+#define COUNTER_DW_RESET_SUPPORT \
+	DT_ANY_INST_HAS_PROP_STATUS_OKAY(resets)
+
+#if COUNTER_DW_RESET_SUPPORT
+#include <zephyr/drivers/reset.h>
+#endif
```

Comment:
what's the problem with unconditionally including this header?

Reply:
removed the conditional include